Python/optimizer.c

```diff
@@ -763,7 +763,7 @@ translate_bytecode_to_trace(
                                 Py_FatalError("garbled expansion");
                         }
 
-                        if (uop == _POP_FRAME || uop == _RETURN_GENERATOR) {
+                        if (uop == _POP_FRAME || uop == _RETURN_GENERATOR || uop == _YIELD_VALUE) {
```

Comment:
@markshannon Shouldn't this same change be made to optimizer_analysis.c (in get_code())?